external/fv3fit/fv3fit/train_microphysics.py

```diff
@@ -533,6 +560,29 @@ def main(config: TrainConfig, seed: int = 0):
         plot_all_output_sensitivities(std_jacobians)
 
 
+def save(config: TrainConfig):
+    path = config.out_url
+    logger.info(f"saving training and validation data to {path}")
+    train, test = config.open_train_test()
+    path = path.rstrip("/")
+    tf.data.experimental.save(train, f"{path}/train")
+    tf.data.experimental.save(test, f"{path}/test")
+
+    # upload config
+    with put_dir(config.out_url) as tmpdir:
+        with open(os.path.join(tmpdir, "config.yaml"), "w") as f:
+            f.write(config.to_yaml())
+
+
+def main(config: TrainConfig, seed: int = 0):
```

Comment:
Would you consider making the "save only" path a new executable module (e.g. `save_microphysics`), instead of altering the API of train_microphysics to have a mode which behaves completely differently? If there are a bunch of kubernetes workflows that would need to be modified for that I won't insist, but typically we try to make different functions/scripts for different tools, rather than writing one script with multiple tools (unless it's clearly an entrypoint, like `vcmtools save` vs `vcmtools train`).

Reply:
I did consider a separate subcommand, but we do indeed have k8s tooling around this end point already...and it's nice to have everything in the yaml.

Reply:
Do you have a way to avoid the ignored-filters bug with one endpoint? It seems like if you do a save step and a train step and for both give it a configuration that says to apply these filters, you necessarily have to apply the filters twice, or have one of these endpoints ignore the configuration option.

Reply:
applying filters twice is ok IMO, since the filter isn't slow and filtering is idempotent.

TBH. I see where you are coming from, but I felt this was a relatively unobstrusive change that didn't break anything and has the benefit of keeping the config declarative, and I am fine taking on the tech debt since I'd like to switch to fv3fit.train anyway after this project is over anyway.

Reply:
I hadn't realized that filters only select data, thought it was an abstract term for "transforms". Should be fine, then. Thanks for walking me through it!